In [107]:
import pandas as pd

In [12]:
begInv = pd.read_csv("BegInvFINAL12312016Sample.csv")

# Deliverable 1A

In [35]:
begInv = begInv.rename(columns = {'onHand':'01_01_2016_Qty'})
begInv['InvCost01_01_2016'] = begInv['01_01_2016_Qty'] * begInv['Price']
begInv_Store = begInv[['Store','01_01_2016_Qty','InvCost01_01_2016']].groupby('Store').sum()
begInv_Store.sort_values(by='InvCost01_01_2016',ascending=False).head(10)
begInv_Store.sort_values(by='InvCost01_01_2016').tail(10).sort_values(by='InvCost01_01_2016', ascending = False)

01_01_2016_Qty  InvCost01_01_2016
Store                                   
34             153852         3291170.24
73             162551         3142497.36
67             158996         3079578.63
66             149314         2973033.90
76             140208         2952418.44
69             144255         2946726.65
38             114368         2232698.77
55             119641         2001263.66
50              94720         1649808.22
79              95330         1503149.48

# Deliverable 1B

In [41]:
bebegInv = begInv.rename(columns = {'onHand':'01_01_2016_Qty'})
begInv['InvCost01_01_2016'] = begInv['01_01_2016_Qty'] * begInv['Price']
begInv_Brand = begInv[['Brand','01_01_2016_Qty','InvCost01_01_2016']].groupby('Brand').sum()
begInv_Brand.sort_values(by='InvCost01_01_2016',ascending=False).head(10)
begInv_Brand.sort_values(by='InvCost01_01_2016').tail(10).sort_values(by='InvCost01_01_2016', ascending = False)

01_01_2016_Qty  InvCost01_01_2016
Brand                                   
3545            14499          463823.01
1233            12016          432455.84
8068            15341          383371.59
4261            15499          340823.01
3858            13649          327439.51
2753             4625          286703.75
8082             9287          278517.13
8680             7066          275503.34
2589             6766          270572.34
3876            14829          266773.71

# Deliverable 2A

In [26]:
endInv = pd.read_csv("EndInvFINAL12312016.csv")
endInv = endInv.rename(columns = {'onHand':'EOY16Qty'})
endInv['EOY16Value'] = endInv['EOY16Qty'] * endInv['Price']
endInv_Store = endInv[['Store','EOY16Qty','EOY16Value']].groupby('Store').sum()
endInv_Store.sort_values(by='EOY16Qty',ascending=False).head(10)
endInv_Store.sort_values(by='EOY16Value').tail(10).sort_values(by='EOY16Value', ascending = False)

EOY16Qty  EOY16Value
Store                      
50       260717  4887260.68
73       164589  3254662.81
67       163765  3076114.82
34       145829  3074616.75
76       143866  2975945.18
69       150848  2968678.82
66       144579  2860504.99
74       166015  2803645.13
38       129397  2463906.85
55       125584  2234836.35

# Deliverable 2B

In [25]:
endInv = pd.read_csv("EndInvFINAL12312016.csv")
endInv = endInv.rename(columns = {'onHand':'EOY16Qty'})
endInv['EOY16Value'] = endInv['EOY16Qty'] * endInv['Price']
endInv_Brand = endInv[['Brand','EOY16Qty','EOY16Value']].groupby('Brand').sum()
endInv_Brand.sort_values(by='EOY16Value',ascending=False).head(10)
endInv_Brand.sort_values(by='EOY16Value').tail(10).sort_values(by='EOY16Value', ascending = False)

EOY16Qty  EOY16Value
Brand                      
1233      15047   526494.53
3545      16770   502932.30
2753       7849   470861.51
8068      15608   366631.92
3405      12268   355649.32
4261      16769   351981.31
2757      11603   336370.97
2589       7922   300956.78
1376      13180   276648.20
2585      10487   272557.13

# Deliverable 3

In [16]:
merge3 = pd.merge(left = begInv, right = endInv, how = 'outer', on = ['InventoryId'])

In [46]:
Inventory = merge3[["InventoryId", "01_01_2016_Qty_x", "InvCost01_01_2016_x", "01_01_2016_Qty_y", "InvCost01_01_2016_y"]]

In [47]:
Inventory.head()

InventoryId  01_01_2016_Qty_x  InvCost01_01_2016_x  01_01_2016_Qty_y  \
0  1_HARDERSFIELD_58               8.0               103.92              11.0   
1  1_HARDERSFIELD_60               7.0                76.93               NaN   
2  1_HARDERSFIELD_62               6.0               221.94               7.0   
3  1_HARDERSFIELD_63               3.0               116.97               7.0   
4  1_HARDERSFIELD_72               6.0               209.94               4.0   

   InvCost01_01_2016_y  
0               142.89  
1                  NaN  
2               258.93  
3               272.93  
4               139.96

In [48]:
len(Inventory)

256042

# Deliverable 4a

In [54]:
Invoices = pd.read_csv("InvoicePurchases12-31-16Sample.csv")

In [55]:
Purchases = pd.read_csv("PurchasesFINAL12-31-16Sample.csv")

In [65]:
vendor_dollars = Purchases.groupby('VendorName')['Dollars'].sum()

In [66]:
top_vendors = vendor_dollars.sort_values(ascending=False)

In [67]:
top_vendors.head(10)

VendorName
DIAGEO NORTH AMERICA INC       21315299.75
MARTIGNETTI COMPANIES          11709222.19
JIM BEAM BRANDS COMPANY        10232521.65
PERNOD RICARD USA              10224899.93
BACARDI USA INC                 7420630.63
CONSTELLATION BRANDS INC        6584032.72
BROWN-FORMAN CORP               5719375.62
ULTRA BEVERAGE COMPANY LLP      5545766.29
E & J GALLO WINERY              5177067.61
M S WALKER INC                  4647908.17
Name: Dollars, dtype: float64

# Deliverable 4b

In [75]:
vendor_freight = Invoices.groupby('VendorName')['Freight'].sum()

In [76]:
top_vendors_freight = vendor_freight.sort_values(ascending=False)

In [77]:
top_vendors_freight.head(10)

VendorName
DIAGEO NORTH AMERICA INC       257032.07
MARTIGNETTI COMPANIES          144719.92
JIM BEAM BRANDS COMPANY        123880.97
PERNOD RICARD USA              123780.22
BACARDI USA INC                 89286.27
CONSTELLATION BRANDS INC        79528.99
BROWN-FORMAN CORP               68601.68
ULTRA BEVERAGE COMPANY LLP      68054.70
E & J GALLO WINERY              61966.91
M S WALKER INC                  55551.82
Name: Freight, dtype: float64

# Deliverable 4c

In [79]:
VendorInfo = Invoices.groupby('VendorName').agg({'Dollars': 'sum', 'Freight': 'sum'})

In [83]:
VendorInfo = VendorInfo[VendorInfo['Dollars'] >= 250000]

In [87]:
Freight_per_dollar = VendorInfo['Freight'] / VendorInfo['Dollars']

In [88]:
sorted_vendors = VendorInfo.sort_values(by='Freight_per_dollar', ascending=False)

In [89]:
sorted_vendors.head(10)

Dollars    Freight  Freight_per_dollar  \
VendorName                                                                
WESTERN SPIRITS BEVERAGE CO    361249.21    1933.19            0.005351   
DIAGEO CHATEAU ESTATE WINES   1365472.83    7259.75            0.005317   
FREDERICK WILDMAN & SONS       759449.24    3999.93            0.005267   
STATE WINE & SPIRITS          1529682.04    8014.98            0.005240   
FLAG HILL WINERY & VINEYARD    300403.20    1573.31            0.005237   
BANFI PRODUCTS CORP           1628866.68    8510.41            0.005225   
MARTIGNETTI COMPANIES        27821473.91  144719.92            0.005202   
STE MICHELLE WINE ESTATES     3086650.70   15919.70            0.005158   
WINE GROUP INC                5258636.79   27100.41            0.005154   
ULTRA BEVERAGE COMPANY LLP   13210613.93   68054.70            0.005152   

                             Freight/Dollar  
VendorName                                   
WESTERN SPIRITS BEVERAGE CO        0.005351  
DIAGEO CHATEAU ESTATE WINES        0.005317  
FREDERICK WILDMAN & SONS           0.005267  
STATE WINE & SPIRITS               0.005240  
FLAG HILL WINERY & VINEYARD        0.005237  
BANFI PRODUCTS CORP                0.005225  
MARTIGNETTI COMPANIES              0.005202  
STE MICHELLE WINE ESTATES          0.005158  
WINE GROUP INC                     0.005154  
ULTRA BEVERAGE COMPANY LLP         0.005152

# Deliverable 4d

In [121]:
transactions = Invoices[(Invoices['Freight'] > 100) & (Invoices['Quantity'] <= 1000)]

In [124]:
Invoices[(Invoices['Freight'] > 100) & (Invoices['Quantity'] <= 1000)].head()

VendorNumber                   VendorName InvoiceDate  PONumber  \
11          1485  CASTLE BRANDS CORP.          2016-01-08      8152   
16          2242  DELICATO VINEYARDS INC       2016-01-06      8139   
19          2555  DISARONNO INTERNATIONAL LLC  2016-01-11      8192   
23          2561  EDRINGTON AMERICAS           2016-01-08      8175   
28          3924  HEAVEN HILL DISTILLERIES     2016-01-08      8155   

        PODate     PayDate  Quantity  Dollars  Freight Approval  
11  2015-12-23  2016-02-19       320  5420.41   179.26      NaN  
16  2015-12-22  2016-02-10       808  6646.46   127.05      NaN  
19  2015-12-25  2016-02-17       385  3506.41   146.86      NaN  
23  2015-12-24  2016-02-10       136  5645.24   218.18      NaN  
28  2015-12-23  2016-02-15       818  7079.02   200.02      NaN

In [136]:
vendor_stats = transactions.groupby('VendorName').agg({'Freight': 'sum', 'Quantity': 'sum', 'Dollars': 'sum'})
transactions.groupby('VendorName').agg({'Freight': 'sum', 'Quantity': 'sum'})

Freight  Quantity
VendorName                                    
CALEDONIA SPIRITS INC         109.80       576
CASTLE BRANDS CORP.           179.26       320
DELICATO VINEYARDS INC        127.05       808
DISARONNO INTERNATIONAL LLC   146.86       385
EDRINGTON AMERICAS           2944.67     18120
HEAVEN HILL DISTILLERIES      200.02       818
KOBRAND CORPORATION           185.93       427
LAIRD & CO                    100.37       982
LUXCO INC                     196.61       819
MAJESTIC FINE WINES           218.05       875
MHW LTD                       177.50       412
PALM BAY INTERNATIONAL INC    242.73       715
REMY COINTREAU USA INC        348.07       382
SIDNEY FRANK IMPORTING CO     133.83       379
STATE WINE & SPIRITS          303.70       541
STOLI GROUP,(USA) LLC         349.02       975
Serralles Usa LLC             122.24       733
WESTERN SPIRITS BEVERAGE CO   169.46       795

In [129]:
freight_quantity = transactions.groupby('VendorName')['Freight'].sum()


In [130]:
freight_quantity[freight_quantity == freight_quantity.max()]

VendorName
EDRINGTON AMERICAS             2944.67
Name: Freight, dtype: float64

## I am confused about my answer to this. It does not match the check figure. However, when I listed out the individual transactions, there were some from companies not in the check figure that fit the criteria. For example, Castle Brands Corp. had a freight of 179.26 and a Quantity of 320.

# Deliverable 4e

In [159]:
fourE = freight_quantity['Freight_per_dollar'] = Invoices['Freight'] / Invoices['Dollars']
fourE = freight_quantity['Freight_per_unit'] = Invoices['Freight'] / Invoices ['Quantity']

In [160]:
print(fourE)

0       0.578333
1       0.571333
2       0.922000
3       0.290614
4       0.221809
          ...   
5538    0.095556
5539    0.040394
5540    0.094735
5541    0.033117
5542    0.037986
Length: 5543, dtype: float64


# Deliverable 5

## Top 10 vendors by total sales

In [171]:
vendor_total_sales = Invoices.groupby('VendorName')['Dollars'].sum()

In [177]:
top_10_vendor_total_sales = vendor_total_sales.sort_values(ascending=False)

In [178]:
top_10_vendor_total_sales.head(10)

VendorName
DIAGEO NORTH AMERICA INC       50959796.85
MARTIGNETTI COMPANIES          27821473.91
JIM BEAM BRANDS COMPANY        24203151.05
PERNOD RICARD USA              24124091.56
BACARDI USA INC                17624378.72
CONSTELLATION BRANDS INC       15573917.90
BROWN-FORMAN CORP              13529433.08
ULTRA BEVERAGE COMPANY LLP     13210613.93
E & J GALLO WINERY             12289608.09
M S WALKER INC                 10935817.30
Name: Dollars, dtype: float64